### PR TITLE
fix: pin versions, add hadolint and remove old 20.04 image

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
## What?

1. Remove old unused ubuntu 20.04 docker image - it's not required anymore
2. Pin apt package versions inside the Docker container to never have an issue that some package is updated to an incompatible version in the background
3. Add `hadolint` linter
4. Small CI simplifications

## Why?

To make sure we don't have any failures using `latest` tag of the Docker image when the Docker image is updated.

### CODE REVIEW

File: .github/workflows/handle_llvm_runner_image.yml [hunk 1]
```diff
@@ -9,53 +9,22 @@ on:
       - main
 
 jobs:
-  check_changes:
```
Comment (antonbaliasnikov) [inline .github/workflows/handle_llvm_runner_image.yml:12]: As we have only one image, this is not required anymore in the first place. Secondly, it was not working as it should have, and anyway, 2 images were built even if only one changed. I'll return it back if required later.

File: .github/workflows/handle_llvm_runner_image.yml [hunk 2]
```diff
@@ -9,53 +9,22 @@ on:
...
-            echo "image_to_build=none" >> "$GITHUB_OUTPUT"
-          fi
-
   handle_image:
-    needs: check_changes
-    if: needs.check_changes.outputs.image_to_build != 'none'
-    runs-on: [matterlabs-ci-runner]
+    runs-on: matterlabs-ci-runner
     strategy:
       matrix:
```
Comment (antonbaliasnikov) [inline .github/workflows/handle_llvm_runner_image.yml:15]: Keep the matrix job to be able to add new images later on without major CI changes. `tag_prefix` equals to image name to simplify the process.

**Comment by antonbaliasnikov** [comment]
@hedgar2017 , could you, please, review this PR when you have some time, thanks in advance!

**Comment by hedgar2017** [comment]
@antonbaliasnikov LGTM, feel free to merge if these tools aren't used anywhere else.

**Pull request merged**